nucypher/blockchain/economics.py

```diff
@@ -167,6 +167,31 @@ def staking_deployment_parameters(self) -> Tuple[int, ...]:
         )
         return tuple(map(int, deploy_parameters))
 
+    def token_supply_function(self, at_period: int) -> int:
+        if at_period < 0:
+            raise ValueError("Period must be a positive integer")
+
+        with localcontext() as ctx:
+            ctx.prec = self._precision
+
+            # Eq. 3 of the mining paper
```

Comment:
Can we have the link to the online version of the Eq.?

Reply:
✔️